src/main/java/org/apache/sling/testing/clients/SlingClientConfig.java

```diff
@@ -171,8 +170,8 @@ public Builder setUrl(String url) throws URISyntaxException {
         public Builder setUrl(URI url) {
             this.url = url;
             // Add / as path if none is present
-            if (Strings.isNullOrEmpty(this.url.getPath()) || !this.url.getPath().endsWith("/")) {
-                this.url = this.url.resolve(Strings.nullToEmpty(this.url.getPath()) + "/");
+            if (StringUtils.isEmpty(this.url.getPath()) || !this.url.getPath().endsWith("/")) {
+                this.url = this.url.resolve((this.url.getPath() == null ? "" : this.url.getPath()) + "/");
```

Comment:
@andreituicu can you use optionals instead?
```suggestion
                this.url = this.url.resolve(Optional.ofNullable(this.url.getPath()).orElse("") + "/");
```